### ISSUE TITLE
Make some default mint coins restrictions

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/cosmos-sdk#49]
## Summary

We need a mintcoins restriction function creator, for easily creating restrictions to only mint coins with the provided prefix. E.g.

```
// NewPrefixMintCoinsRestriction creates a mint coins restriction fn, that only allows banks to mint coins that have a denom beginning with "...."
func NewPrefixMintCoinsRestriction(prefix string) MintCoinsRestrictionFn {
...
}
```

## Problem Definition

We're going to use this same code for gamm and token factory.

Blocked on #48 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by mattverse** [comment]
Should I create a PR against https://github.com/osmosis-labs/cosmos-sdk/pull/50 branch for this issue?What Im thinking as of now is adding a default mintcoin restriction function creator together with https://github.com/osmosis-labs/cosmos-sdk/pull/50 and get it merged and reviewed together.

**Comment by mattverse** [comment]
Also do we have an idea on what default minting restriction we want? I'm currently thinking of making a variadic function taking in strings that you would want to restrict minting on

**Comment by ValarDragon** [comment]
oh interesting idea, I think that makes sense. Basically emits a restriction fn, that blocks emitting anything that starts with any of the variable inputs.

**Comment by ValarDragon** [comment]
Closed in #57
